### PR TITLE
RegisterTMRequest unit test class

### LINKED ISSUE #189: Add unit test to fescar-core module
Add unit test to fescar-core module

### PR DESCRIPTION
**Pull request by Danaykap**
### Ⅰ. Describe what this PR did
This pr enhances the test coverage. The class added tests RegisterTMRequest class

### Ⅱ. Does this pull request fix one issue?
fixes #189

### CODE REVIEW

File: core/src/test/java/io/seata/core/protocol/RegisterTMRequestTest.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
+package io.seata.core.protocol;
+
+import org.junit.jupiter.api.*;
```
Comment (xingfudeshi) [inline core/src/test/java/io/seata/core/protocol/RegisterTMRequestTest.java]: please import explicit class type.

File: core/src/test/java/io/seata/core/protocol/RegisterTMRequestTest.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
+package io.seata.core.protocol;
```
Comment (xingfudeshi) [inline core/src/test/java/io/seata/core/protocol/RegisterTMRequestTest.java]: please add copyright header.

File: core/src/test/java/io/seata/core/protocol/RegisterTMRequestTest.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+	/** Constructor with arguments **/
+	@BeforeAll
+	public static void setupWithValues() {
+		registerTMRequest = new RegisterTMRequest(APP_ID, TSG, ED);
+	}
+
+	/** Test get type code **/
+	@Test
+	public void testGetTypeCode() {
+		Assertions.assertEquals(101, registerTMRequest.getTypeCode());
```
Comment (xingfudeshi) [inline core/src/test/java/io/seata/core/protocol/RegisterTMRequestTest.java]: please define a constant for the value "101"

**Comment by kaitithoma** [comment]
> Can you add test cases for other methods of this class? Such as encode and decode methods

We could try it! Should we create a new pr for this purpose?

**Comment by xingfudeshi** [comment]
> > Can you add test cases for other methods of this class? Such as encode and decode methods
> 
> We could try it! Should we create a new pr for this purpose?

it's better to add the other tests of this class in the same PR.

**Comment by kaitithoma** [comment]
> > > Can you add test cases for other methods of this class? Such as encode and decode methods
> > 
> > 
> > We could try it! Should we create a new pr for this purpose?
> 
> it's better to add the other tests of this class in the same PR.

encode and decode methods are in AbstractIdentifyRequest class but we intended to test RegisterTMRequest which extends it. Aren' t they supposed to be different classes?

**Comment by xingfudeshi** [comment]
Each test class should correspond to a concrete class. RegisterTMRequest, although it inherits from AbstractIdentifyRequest, overrides the decode and encode methods, so the unit test case here is only for RegisterTMRequest.

**Comment by kaitithoma** [comment]
We added test methods for decoding. Sorry if there are unnecessary code lines and mistakes but we aren' t very familiar to ByteBuf variables. Thank you!

**Comment by xingfudeshi** [comment]
No worries.we can help you improve the code.

**Pull request merged**